recipes/gtsam/all/conanfile.py

```diff
@@ -0,0 +1,135 @@
+               "TYPEDEF_POINTS_TO_VECTORS": [True, False],
+               "SUPPORT_NESTED_DISSECTION": [True, False],
+               "TANGENT_PREINTEGRATION": [True, False],
+               "BUILD_WRAP": [True, False],
+               "WRAP_SERIALIZATION": [True, False],
+               "BUILD_UNSTABLE": [True, False],
+               "DISABLE_NEW_TIMERS": [True, False],
+               "BUILD_TYPE_POSTFIXES": [True, False],
+               "Boost_USE_STATIC_LIBS": [True, False],
+               "INSTALL_MATLAB_TOOLBOX": [True, False],
+               "INSTALL_CYTHON_TOOLBOX": [True, False],
+               "INSTALL_CPPUNITLITE": [True, False],
+               "INSTALL_GEOGRAPHICLIB": [True, False]}
+
+    default_options = {"shared": False,
+                       "fPIC": True,
+                        "USE_QUATERNIONS": False,
+                        "POSE3_EXPMAP": False,
+                        "ROT3_EXPMAP": False,
+                        "ENABLE_CONSISTENCY_CHECKS": False,
+                        "WITH_TBB": True,
+                        "WITH_EIGEN_MKL": False,
+                        "WITH_EIGEN_MKL_OPENMP": False,
+                        "THROW_CHEIRALITY_EXCEPTION": True,
+                        "ALLOW_DEPRECATED_SINCE_V4": True,
+                        "TYPEDEF_POINTS_TO_VECTORS": False,
+                        "SUPPORT_NESTED_DISSECTION": False,
+                        "TANGENT_PREINTEGRATION": False,
+                        "BUILD_WRAP": True,
+                        "WRAP_SERIALIZATION": True,
+                        "BUILD_UNSTABLE": True,
+                        "DISABLE_NEW_TIMERS": False,
+                        "BUILD_TYPE_POSTFIXES": True,
+                        "Boost_USE_STATIC_LIBS": True,
+                        "INSTALL_MATLAB_TOOLBOX": False,
+                        "INSTALL_CYTHON_TOOLBOX": False,
+                        "INSTALL_CPPUNITLITE": True,
+                        "INSTALL_GEOGRAPHICLIB": False}
+    generators = "cmake"
+    exports_sources = ["patches/*","CMakeLists.txt"]
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+    _cmake = None
+
+    def _configure_cmake(self):
+        if not self._cmake:
+            self._cmake = CMake(self)
+            self._cmake.verbose = False
+            self._cmake.definitions["BUILD_SHARED_LIBS"] = self.options.shared
+            self._cmake.definitions["GTSAM_USE_QUATERNIONS"] = self.options.USE_QUATERNIONS
+            self._cmake.definitions["GTSAM_POSE3_EXPMAP"] = self.options.POSE3_EXPMAP
+            self._cmake.definitions["GTSAM_ROT3_EXPMAP"] = self.options.ROT3_EXPMAP
+            self._cmake.definitions["GTSAM_ENABLE_CONSISTENCY_CHECKS"] = self.options.ENABLE_CONSISTENCY_CHECKS
+            self._cmake.definitions["GTSAM_WITH_TBB"] = self.options.WITH_TBB
+            self._cmake.definitions["GTSAM_WITH_EIGEN_MKL"] = self.options.WITH_EIGEN_MKL
```

Comment:
I think `mkl` should also be added to the requirements.

Reply:
There is no recipe for MKL yet. I forgot to add the dependency on TBB, I'll upload a fix soon.